tests/core/sensors/test_external_task.py

```diff
@@ -11,147 +11,122 @@
-        sensor.execute(context)
-
-    assert isinstance(exc.value.trigger, DagStateTrigger), "Trigger is not a DagStateTrigger"
-
-
-@mock.patch("astronomer.providers.core.sensors.external_task.ExternalTaskSensorAsync.get_count")
-def test_task_sensor_execute_complete_throws_exc(mocked_count, session, dag, context):
-    """
-    Asserts that the correct exception is raised when not every task monitored by
-    the sensor is executed successfully.
-    """
-    mocked_count.return_value = 0
-    sensor = ExternalTaskSensorAsync(
-        external_task_id=TEST_EXT_TASK_ID,
-        external_dag_id=TEST_EXT_DAG_ID,
-        task_id=TEST_TASK_ID,
-        dag=dag,
-    )
-
-    with pytest.raises(AirflowException) as exc:
-        sensor.execute_complete(context, session)
-
-    assert str(exc.value) == "The external task wait_for_me_task in DAG wait_for_me_dag failed."
-
-
-@mock.patch("astronomer.providers.core.sensors.external_task.ExternalTaskSensorAsync.get_count")
-def test_dag_sensor_execute_complete_throws_exc(mocked_count, session, dag, context):
-    """
-    Asserts that the correct exception is raised when not every DAG monitored by
-    the sensor is executed successfully.
-    """
-    mocked_count.return_value = 0
-    sensor = ExternalTaskSensorAsync(
-        external_dag_id=TEST_EXT_DAG_ID,
-        task_id=TEST_TASK_ID,
-        dag=dag,
-    )
-
-    with pytest.raises(AirflowException) as exc:
-        sensor.execute_complete(context, session)
-
-    assert str(exc.value) == "The external DAG wait_for_me_dag failed."
-
-
-def test_get_execution_dates(dag, context):
-    """
-    Asserts that this helper function returns execution_dates as expected
-    depending on whether execution_delta, execution_date_fn, or neither
-    argument is provided.
-    """
-    sensor = ExternalTaskSensorAsync(
-        external_dag_id=TEST_EXT_DAG_ID,
-        task_id=TEST_TASK_ID,
-        dag=dag,
-    )
-
-    # Case #1 sensor.execution_delta only
-    sensor.execution_delta = timedelta(days=1)
-    execution_dates = sensor.get_execution_dates(context)
-    assert execution_dates[0] == datetime(2014, 12, 31)
-    assert len(execution_dates) == 1
-    sensor.execution_delta = None  # Reset this field for next test
-
-    # Case #2 sensor.execution_date_fn only
-    sensor.execution_date_fn = lambda dt: [dt + timedelta(days=i) for i in range(2)]
-    execution_dates = sensor.get_execution_dates(context)
-    assert execution_dates[0] == datetime(2015, 1, 1)
-    assert execution_dates[1] == datetime(2015, 1, 2)
-    assert len(execution_dates) == 2
-    sensor.execution_date_fn = None  # Reset this field for next test
-
-    # Case #3 both sensor.execution_delta and sensor.execution_date_fn are set to None
-    execution_dates = sensor.get_execution_dates(context)
-    assert execution_dates[0] == datetime(2015, 1, 1)
-    assert len(execution_dates) == 1
+
+class TestExternalTaskSensorAsync:
+    TASK_ID = "external_task_sensor_check"
```

Comment:
can we add this variables inside setup_class like class variables?

Reply:
do you mean I should create a setup class and keep this variable in that class? Can you please elaborate a little bit?

Reply:
Yes you do like this 

```
class Test:
    def setup_class(self) -> None:
        self.xxxx: str = ""
        self.yyyyy: str = ''
        self.connection_string = (
        )

```

Reply:
or you can create def test_init() inside the class

Reply:
these variables are just single lines and getting used in this class only so moving outside to a different class not sure how much value will add.  regarding variables like conn which might get used in multiple test classes, I'm planning to keep them in `util`